### ISSUE TITLE
PrinterState() suite failures

### ISSUE THREAD

**Issue by marctan** [google/cloudprint_logocert#81]
We only have 2 failures before namely: "testNoMediaIntray" and "testOpenPaperTray" because our printer is in warning state but the testscript is expecting us to be in error state if there's no media or if tray is open. @YasuhiroFuruta implement a fix for this at #53 and #66 but after running it, we encountered the errors below:

testNoMediaInTray	- Failed - state message "Input tray is empty (Tray 1 Empty)" is not allowed

testCoverOpen - Failed - state message "Front door is open (Close door or insert cartridge)" is not allowed

testOpenPaperTray - Failed - state message "Input tray is open (Tray 1 Missing)" is not allowed		
	
testPaperJam - Failed - state message "Paper jam (Paper jam, rear door B. [202.15C])" is not allowed	
			
testRemoveTonerCartridge - Failed - state message "Black toner is removed (Reinstall missing or unresponsive cartridge [31.40C])" is not allowed						

testExhaustTonerCartridge - Failed - state message "Black toner is empty (Replace cartridge, 0 estimated pages remain [88.30K])" is not allowed				
		
testReplaceMissingToner - Failed - state message "Black toner level is 92% – 17600 pages remaining" is not allowed					

Please have this check on your end. 
Thanks!

**Comment by kdlucas** [comment]
Yasuhiro-san,

Do you have any ideas about why we're seeing "is not allowed" here? I think we need to alter the code a bit.

**Comment by YasuhiroFuruta** [comment]
Changes of #66 confirm the tails of state massages.
I think @marctan's printer sends the "vendor_message" fields on CDS and the server append the "vendor_message" value to the tails of the state messages with surrounding by "()".
Also, @marctan's printer sends the "level_pages" fields and the server append the "level_pages" value to the tails of the state messages.

I will fix it by removing the parts of the state messages surrounded by "()"  before confirming the tail and allowing the "level_pages" value in the state messages.

**Comment by YasuhiroFuruta** [comment]
Hi @marctan,

I fixed this issue, and pushed to "fix_prt_state" branch (fddba9d509172e1392d610265324a24d680589c1).
Please test it with your printer.
(I tested it with using "Google Cloud Print API Simulation" page.)

If your printer is passed the test cases, I will create PR.

**Comment by marctan** [comment]
I'll test this and will provide feedback

Thanks!

**Comment by marctan** [comment]
Hi, @YasuhiroFuruta 

All passed in PrinterState suite. Except for **testExhaustTonerCartridge**

Logs below:

> LogoCert INFO: test_id: b73b5b6b-9398-48ad-9646-dbb501b32f8c: Failed
> LogoCert INFO: b73b5b6b-9398-48ad-9646-dbb501b32f8c: testExhaustTonerCartridge
> LogoCert INFO: b73b5b6b-9398-48ad-9646-dbb501b32f8c: required suffix " is empty"
>  is not in state messages

Here's the message of our printer at the Management Page:
![gcpscript](https://cloud.githubusercontent.com/assets/19194749/18904575/4c5003ca-8591-11e6-874a-cc5b3f93565e.png)


Also I've noticed in **testPaperJam()**, I think we should pass "test_id2 and test_name2" in the parameter of the 2nd VerifyStateMessages() instead of test_id and testname. This is only for user clarity. 

Thanks!

**Comment by YasuhiroFuruta** [comment]
For testExhaustTonerCartridge():
I think your printer sends "OK" state in "marker_state".
The printer should send 'EXHAUSTED' state in "marker_state" in this testcase.

For testPaperJam():
Exactly, I will fix it.
Thank you.

**Comment by marctan** [comment]
For testExhaustTonerCartridge():
Our printer sent EXHAUSTED state.

I've read the logic of the script and i think I know where it failed at (correct me if I'm wrong)

starting at Line 3089 of testcert.py:
```
3089:    for message in messages:
3090:      if required_suffix and message.endswith(required_suffix):
3091:        found = True
3092:      else:
3093:        if not message.endswith(allowed_suffixes):
3094:          notes = 'state message "%s" is not allowed' % message
3095:          self.LogTest(test_id, test_name, 'Failed', notes)
3096:          return False
```

Our printer message is "Black toner level is low (0%) - 0 pages remaining." So it will enter the else statement at line 3092 because the **message** does not end with "is empty".

It will also not enter line 3093 since the message is ending with one of the lists in "allowed_suffixes", so I think we should add an else statement and set "found = true" if message is ending with one of the lists in allowed suffixes. I'm not sure with this so please verify this on your end. 

Thanks!

**Comment by YasuhiroFuruta** [comment]
Hi @marctan,

Would you get CDS of your printer in marker exhaust state by https://www.google.com/cloudprint/simulate.html, please?

**Comment by marctan** [comment]
Hi @YasuhiroFuruta 

I've rechecked the test again and found out that the toner I used is not actually empty. It is showing 0 pages remaining but its state is not EMPTY. So I've tried looking around for a toner that is really EMPTY. And luckily I was able to borrow one from the other team.  I've re run testExhaustTonerCartridge() and it is now **PASSING**. Sorry for the wrong info. :( 

Thank you very much for your time.
